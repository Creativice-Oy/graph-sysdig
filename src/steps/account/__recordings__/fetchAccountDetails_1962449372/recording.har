{
  "log": {
    "_recordingName": "fetchAccountDetails",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "0d29cdc1ea9a8e8f7ab793f1e65b94a7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.us4.sysdig.com"
            }
          ],
          "headersSize": 285,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.us4.sysdig.com/api/user/me"
        },
        "response": {
          "bodySize": 2675,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 2675,
            "text": "{\"user\":{\"termsAndConditions\":true,\"timezone\":\"+00:00\",\"pictureUrl\":\"http://www.gravatar.com/avatar/ba5b87ffa4527a556b5f6da95fe36fe5\",\"customerSettings\":{\"sysdig\":{\"enabled\":false,\"enabledSSE\":false,\"buckets\":[]},\"plan\":{\"maxAgents\":15,\"onDemandAgents\":0,\"maxTeams\":-1,\"timelines\":[{\"from\":null,\"to\":null,\"sampling\":10000000},{\"from\":null,\"to\":null,\"sampling\":60000000},{\"from\":null,\"to\":null,\"sampling\":600000000},{\"from\":null,\"to\":null,\"sampling\":3600000000},{\"from\":null,\"to\":null,\"sampling\":86400000000}],\"metricsSettings\":{\"enforce\":false,\"showExperimentals\":false,\"limits\":{\"jmx\":500,\"statsd\":1000,\"appCheck\":500,\"prometheus\":8000,\"prometheusPerProcess\":500,\"connections\":80,\"progAggregationCount\":12,\"appCheckAggregationCount\":12,\"promMetricsWeight\":0.0,\"topFilesCount\":10,\"topDevicesCount\":10,\"hostServerPorts\":10,\"containerServerPorts\":5,\"limitKubernetesResources\":false,\"kubernetesPods\":10000,\"kubernetesJobs\":10000,\"containerDensity\":200,\"meerkatSuited\":true},\"enforceAgentAggregation\":true,\"enablePromCalculatedIngestion\":false,\"enablePromScrapeV2\":true,\"enableGlobalK8sAllPodContainers\":false},\"secureEnabled\":true,\"monitorEnabled\":false,\"allocatedAgentsCount\":15,\"subscriptionState\":\"trialing\",\"paymentsIntegrationId\":{\"id\":\"not present\",\"ttl\":{\"ttl\":43}},\"pricingPlan\":\"pro\",\"indirectCustomer\":false,\"trialPlanName\":\"secure-enterprise-standard-monthly\",\"partner\":\"Direct\",\"trialEndDate\":1664383443236,\"subscriptionVersion\":2,\"migratedToV2Direct\":false,\"overageAssessmentEligible\":false},\"environment\":{}},\"customer\":{\"id\":5001686,\"name\":null,\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"externalId\":\"fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9\",\"dateCreated\":1661791442836},\"oauth\":false,\"agentInstallParams\":{\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"collectorAddress\":\"ingest.us4.sysdig.com\",\"collectorPort\":6443,\"checkCertificate\":true,\"sslEnabled\":true},\"remoteWriteParams\":{\"serverAddress\":\"app.us4.sysdig.com\"},\"properties\":{\"has_been_invited\":false},\"resetPassword\":false,\"additionalRoles\":[],\"teamRoles\":[{\"teamId\":50006626,\"teamName\":\"Secure Operations\",\"teamTheme\":\"#7BB0B2\",\"userId\":50003375,\"userName\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"ROLE_TEAM_MANAGER\",\"admin\":true},{\"teamId\":50006625,\"teamName\":\"Monitor Operations\",\"teamTheme\":\"#7BB0B2\",\"userId\":50003375,\"userName\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"ROLE_TEAM_MANAGER\",\"admin\":true}],\"lastUpdated\":1663493736587,\"dateActivated\":1661791669709,\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"intercomUserIdHash\":\"7941192844d448aedd650df4bd26799efd5fca275a1f1eef775ee67b6cf061ad\",\"uniqueIntercomUserId\":\"50003375.fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9\",\"currentTeam\":50006626,\"name\":\"stefan.virag@contractor.jupiterone.com\",\"id\":50003375,\"version\":2,\"enabled\":true,\"firstName\":\"Stefan\",\"lastName\":\"Virag\",\"systemRole\":\"ROLE_CUSTOMER\",\"username\":\"stefan.virag@contractor.jupiterone.com\",\"status\":\"confirmed\",\"products\":[\"SDC\",\"SDS\"],\"dateCreated\":1661791442936,\"lastSeenOnSecure\":1663493736587}}"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Sun, 18 Sep 2022 09:46:17 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding, User-Agent"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 498,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-09-18T09:46:16.712Z",
        "time": 518,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 518
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
