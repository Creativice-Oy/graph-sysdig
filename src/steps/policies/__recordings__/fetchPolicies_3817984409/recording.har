{
  "log": {
    "_recordingName": "fetchPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "0d29cdc1ea9a8e8f7ab793f1e65b94a7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.us4.sysdig.com"
            }
          ],
          "headersSize": 285,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.us4.sysdig.com/api/user/me"
        },
        "response": {
          "bodySize": 2673,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 2673,
            "text": "{\"user\":{\"termsAndConditions\":true,\"timezone\":\"+00:00\",\"pictureUrl\":\"http://www.gravatar.com/avatar/ba5b87ffa4527a556b5f6da95fe36fe5\",\"customerSettings\":{\"sysdig\":{\"enabled\":false,\"enabledSSE\":false,\"buckets\":[]},\"plan\":{\"maxAgents\":15,\"onDemandAgents\":0,\"maxTeams\":-1,\"timelines\":[{\"from\":null,\"to\":null,\"sampling\":10000000},{\"from\":null,\"to\":null,\"sampling\":60000000},{\"from\":null,\"to\":null,\"sampling\":600000000},{\"from\":null,\"to\":null,\"sampling\":3600000000},{\"from\":null,\"to\":null,\"sampling\":86400000000}],\"metricsSettings\":{\"enforce\":false,\"showExperimentals\":false,\"limits\":{\"jmx\":500,\"statsd\":1000,\"appCheck\":500,\"prometheus\":8000,\"prometheusPerProcess\":500,\"connections\":80,\"progAggregationCount\":12,\"appCheckAggregationCount\":12,\"promMetricsWeight\":0.0,\"topFilesCount\":10,\"topDevicesCount\":10,\"hostServerPorts\":10,\"containerServerPorts\":5,\"limitKubernetesResources\":false,\"kubernetesPods\":10000,\"kubernetesJobs\":10000,\"containerDensity\":200,\"meerkatSuited\":true},\"enforceAgentAggregation\":true,\"enablePromCalculatedIngestion\":false,\"enablePromScrapeV2\":true,\"enableGlobalK8sAllPodContainers\":false},\"secureEnabled\":true,\"monitorEnabled\":false,\"allocatedAgentsCount\":15,\"subscriptionState\":\"trialing\",\"paymentsIntegrationId\":{\"id\":\"not present\",\"ttl\":{\"ttl\":43}},\"pricingPlan\":\"pro\",\"indirectCustomer\":false,\"trialPlanName\":\"secure-enterprise-standard-monthly\",\"partner\":\"Direct\",\"trialEndDate\":1664383443236,\"subscriptionVersion\":2,\"migratedToV2Direct\":false,\"overageAssessmentEligible\":false},\"environment\":{}},\"customer\":{\"id\":5001686,\"name\":null,\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"externalId\":\"fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9\",\"dateCreated\":1661791442836},\"oauth\":false,\"agentInstallParams\":{\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"collectorAddress\":\"ingest.us4.sysdig.com\",\"collectorPort\":6443,\"checkCertificate\":true,\"sslEnabled\":true},\"remoteWriteParams\":{\"serverAddress\":\"app.us4.sysdig.com\"},\"properties\":{\"has_been_invited\":false},\"resetPassword\":false,\"additionalRoles\":[],\"teamRoles\":[{\"teamId\":50006626,\"teamName\":\"Secure Operations\",\"teamTheme\":\"#7BB0B2\",\"userId\":50003375,\"userName\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"ROLE_TEAM_MANAGER\",\"admin\":true},{\"teamId\":50006625,\"teamName\":\"Monitor Operations\",\"teamTheme\":\"#7BB0B2\",\"userId\":50003375,\"userName\":\"stefan.virag@contractor.jupiterone.com\",\"role\":\"ROLE_TEAM_MANAGER\",\"admin\":true}],\"lastUpdated\":1663575912770,\"dateActivated\":1661791669709,\"accessKey\":\"1be2fe7d-f00f-4513-9a15-1384c35d8497\",\"intercomUserIdHash\":\"7941192844d448aedd650df4bd26799efd5fca275a1f1eef775ee67b6cf061ad\",\"uniqueIntercomUserId\":\"50003375.fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9\",\"currentTeam\":50006626,\"name\":\"stefan.virag@contractor.jupiterone.com\",\"id\":50003375,\"version\":2,\"enabled\":true,\"username\":\"stefan.virag@contractor.jupiterone.com\",\"products\":[\"SDS\",\"SDC\"],\"firstName\":\"Stefan\",\"lastName\":\"Virag\",\"systemRole\":\"ROLE_CUSTOMER\",\"status\":\"confirmed\",\"dateCreated\":1661791442936,\"lastSeenOnSecure\":1663575912770}}"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 19 Sep 2022 08:35:02 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding, User-Agent"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 498,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-09-19T08:35:02.232Z",
        "time": 547,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 547
        }
      },
      {
        "_id": "6fd4e7a6bed8615501f32ff2a2f8486e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "app.us4.sysdig.com"
            }
          ],
          "headersSize": 307,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://app.us4.sysdig.com/api/scanning/policies/v2/policies"
        },
        "response": {
          "bodySize": 1684,
          "content": {
            "mimeType": "application/json",
            "size": 1684,
            "text": "{\"policies\":[{\"id\":17384,\"name\":\"Sysdig Best Practices\",\"identifier\":\"sysdig_best_practices\",\"description\":\"Recommended out of the box image scanning checks.\",\"policyType\":\"alwaysApply\",\"bundles\":[{\"id\":1,\"name\":\"Severe vulnerabilities with a Fix\",\"identifier\":\"severe_vulnerabilities_with_a_fix\",\"description\":\"Fail on Severe vulnerabilities that can be fixed.\",\"bundleType\":\"predefined\",\"rules\":[{\"ruleType\":\"vulnSeverityAndThreats\",\"predicates\":[{\"type\":\"vulnSeverity\",\"extra\":{\"level\":\"critical\"}},{\"type\":\"vulnIsFixable\"}]},{\"ruleType\":\"vulnSeverityAndThreats\",\"predicates\":[{\"type\":\"vulnSeverity\",\"extra\":{\"level\":\"high\"}},{\"type\":\"vulnIsFixableWithAge\",\"extra\":{\"age\":30}}]},{\"ruleType\":\"vulnSeverityAndThreats\",\"predicates\":[{\"type\":\"vulnSeverity\",\"extra\":{\"level\":\"high\"}},{\"type\":\"vulnExploitableViaNetwork\"}]}],\"creationTimestamp\":\"2022-03-08T16:42:57.270495Z\",\"updateTimestamp\":\"2022-03-08T16:42:57.270495Z\"}],\"creationTimestamp\":\"2022-09-02T11:58:59.03062Z\",\"updateTimestamp\":\"2022-09-02T11:58:59.03062Z\"},{\"id\":22461,\"name\":\"My New Pipeline Policy 1\",\"identifier\":\"my-new-pipeline-policy-1\",\"description\":\"My New Pipeline Policy Description 1\",\"policyType\":\"alwaysApply\",\"bundles\":[{\"id\":10803,\"name\":\"My new bundle 1\",\"identifier\":\"my-new-bundle-1\",\"description\":\"My new bundle description 1\",\"bundleType\":\"custom\",\"rules\":[{\"ruleType\":\"vulnSeverityAndThreats\",\"predicates\":[{\"type\":\"vulnSeverity\",\"extra\":{\"level\":\"critical\"}},{\"type\":\"vulnAge\",\"extra\":{\"age\":7}}]}],\"creationTimestamp\":\"2022-09-16T09:52:17.357177Z\",\"updateTimestamp\":\"2022-09-16T09:52:17.357177Z\"}],\"creationTimestamp\":\"2022-09-16T09:53:16.483187Z\",\"updateTimestamp\":\"2022-09-16T09:53:16.483187Z\"}]}\n"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "60"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "57"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1663576512"
            },
            {
              "name": "date",
              "value": "Mon, 19 Sep 2022 08:35:03 GMT"
            },
            {
              "name": "content-length",
              "value": "1684"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 327,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-09-19T08:35:02.834Z",
        "time": 470,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 470
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
