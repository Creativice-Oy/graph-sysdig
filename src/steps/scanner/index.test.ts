import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchScannerDetails } from '.';
import { integrationConfig } from '../../../test/config';
import { setupSysdigRecording } from '../../../test/recording';

describe('#fetchScannerDetails', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupSysdigRecording({
      directory: __dirname,
      name: 'fetchScannerDetails',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchScannerDetails(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const scanner = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('sysdig_scanner'),
    );
    expect(scanner.length).toBe(1);
    expect(scanner).toMatchGraphObjectSchema({
      _class: ['Service'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'sysdig_scanner' },
          name: { type: 'string' },
          username: { type: 'string' },
          displayName: { type: 'string' },
          id: { type: 'string' },
          version: { type: 'number' },
          products: {
            type: 'array',
            items: { type: 'string' },
          },
          category: {
            type: 'array',
            items: { type: 'string' },
          },
          function: {
            type: 'array',
            items: { type: 'string' },
          },
        },
      },
    });
  });
});
