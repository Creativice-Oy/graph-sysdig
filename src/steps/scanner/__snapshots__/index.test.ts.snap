// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchScannerDetails should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Service",
      ],
      "_key": "sysdig_scanner:50003375",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "accessKey": "1be2fe7d-f00f-4513-9a15-1384c35d8497",
            "additionalRoles": Array [],
            "agentInstallParams": Object {
              "accessKey": "1be2fe7d-f00f-4513-9a15-1384c35d8497",
              "checkCertificate": true,
              "collectorAddress": "ingest.us4.sysdig.com",
              "collectorPort": 6443,
              "sslEnabled": true,
            },
            "currentTeam": 50006626,
            "customer": Object {
              "accessKey": "1be2fe7d-f00f-4513-9a15-1384c35d8497",
              "dateCreated": 1661791442836,
              "externalId": "fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9",
              "id": 5001686,
              "name": null,
            },
            "customerSettings": Object {
              "environment": Object {},
              "plan": Object {
                "allocatedAgentsCount": 15,
                "indirectCustomer": false,
                "maxAgents": 15,
                "maxTeams": -1,
                "metricsSettings": Object {
                  "enableGlobalK8sAllPodContainers": false,
                  "enablePromCalculatedIngestion": false,
                  "enablePromScrapeV2": true,
                  "enforce": false,
                  "enforceAgentAggregation": true,
                  "limits": Object {
                    "appCheck": 500,
                    "appCheckAggregationCount": 12,
                    "connections": 80,
                    "containerDensity": 200,
                    "containerServerPorts": 5,
                    "hostServerPorts": 10,
                    "jmx": 500,
                    "kubernetesJobs": 10000,
                    "kubernetesPods": 10000,
                    "limitKubernetesResources": false,
                    "meerkatSuited": true,
                    "progAggregationCount": 12,
                    "promMetricsWeight": 0,
                    "prometheus": 8000,
                    "prometheusPerProcess": 500,
                    "statsd": 1000,
                    "topDevicesCount": 10,
                    "topFilesCount": 10,
                  },
                  "showExperimentals": false,
                },
                "migratedToV2Direct": false,
                "monitorEnabled": false,
                "onDemandAgents": 0,
                "overageAssessmentEligible": false,
                "partner": "Direct",
                "paymentsIntegrationId": Object {
                  "id": "not present",
                  "ttl": Object {
                    "ttl": 43,
                  },
                },
                "pricingPlan": "pro",
                "secureEnabled": true,
                "subscriptionState": "trialing",
                "subscriptionVersion": 2,
                "timelines": Array [
                  Object {
                    "from": null,
                    "sampling": 10000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 60000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 600000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 3600000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 86400000000,
                    "to": null,
                  },
                ],
                "trialEndDate": 1664383443236,
                "trialPlanName": "secure-enterprise-standard-monthly",
              },
              "sysdig": Object {
                "buckets": Array [],
                "enabled": false,
                "enabledSSE": false,
              },
            },
            "dateActivated": 1661791669709,
            "dateCreated": 1661791442936,
            "enabled": true,
            "firstName": "Stefan",
            "id": 50003375,
            "intercomUserIdHash": "7941192844d448aedd650df4bd26799efd5fca275a1f1eef775ee67b6cf061ad",
            "lastName": "Virag",
            "lastSeenOnSecure": 1663497618871,
            "lastUpdated": 1663497618871,
            "name": "stefan.virag@contractor.jupiterone.com",
            "oauth": false,
            "pictureUrl": "http://www.gravatar.com/avatar/ba5b87ffa4527a556b5f6da95fe36fe5",
            "products": Array [
              "SDC",
              "SDS",
            ],
            "properties": Object {
              "has_been_invited": false,
            },
            "remoteWriteParams": Object {
              "serverAddress": "app.us4.sysdig.com",
            },
            "resetPassword": false,
            "status": "confirmed",
            "systemRole": "ROLE_CUSTOMER",
            "teamRoles": Array [
              Object {
                "admin": true,
                "role": "ROLE_TEAM_MANAGER",
                "teamId": 50006626,
                "teamName": "Secure Operations",
                "teamTheme": "#7BB0B2",
                "userId": 50003375,
                "userName": "stefan.virag@contractor.jupiterone.com",
              },
              Object {
                "admin": true,
                "role": "ROLE_TEAM_MANAGER",
                "teamId": 50006625,
                "teamName": "Monitor Operations",
                "teamTheme": "#7BB0B2",
                "userId": 50003375,
                "userName": "stefan.virag@contractor.jupiterone.com",
              },
            ],
            "termsAndConditions": true,
            "timezone": "+00:00",
            "uniqueIntercomUserId": "50003375.fbd8b3fc-781c-4e7f-afac-a2ee9309f4c9",
            "username": "stefan.virag@contractor.jupiterone.com",
            "version": 2,
          },
        },
      ],
      "_type": "sysdig_scanner",
      "active": undefined,
      "category": Array [
        "image",
      ],
      "createdOn": undefined,
      "displayName": "Stefan Virag",
      "function": "scanning",
      "id": "50003375",
      "name": "stefan.virag@contractor.jupiterone.com",
      "products": Array [
        "SDC",
        "SDS",
      ],
      "status": "confirmed",
      "username": "stefan.virag@contractor.jupiterone.com",
      "version": 2,
    },
  ],
  "encounteredTypes": Array [
    "sysdig_scanner",
  ],
  "numCollectedEntities": 1,
}
`;
